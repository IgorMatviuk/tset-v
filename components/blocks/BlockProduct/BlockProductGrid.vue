<template>
  <div class="product-grid">
    <div
      v-for="(preview, index) in formattedPreviews"
      :key="index"
      class="product-grid__item"
    >
      <BlockProductPreview :item="preview" />
    </div>
  </div>
</template>

<script>
import BlockProductPreview from './BlockProductPreview'

import formattedPreviews from '@/mixins/formatted-previews'

export default {
  components: {
    BlockProductPreview
  },
  mixins: [formattedPreviews],
  props: {
    previews: {
      type: Array,
      required: true
    }
  }
}
</script>

<style lang="scss">
.product-grid {
  display: none;

  &__item {
    &:not(:last-of-type) {
      margin-bottom: 6px;
    }
  }

  @include media-breakpoint-up(md) {
    display: block;
  }

  @include media-breakpoint-up(lg) {
    display: flex;
    flex-wrap: wrap;
    margin-left: -2.5px;

    &__item {
      flex: 0 0 50%;
      max-width: 50%;
      padding-right: 2.5px;
      padding-left: 2.5px;
    }
  }

  @include media-breakpoint-up(xl) {
    margin-left: -4px;

    &__item {
      padding-right: 4px;
      padding-left: 4px;

      &:not(:last-of-type) {
        margin-bottom: 8px;
      }
    }
  }
}
</style>
