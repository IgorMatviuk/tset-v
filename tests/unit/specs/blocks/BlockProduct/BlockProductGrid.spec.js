import { shallowMount } from '@vue/test-utils'

import BlockProductGrid from '@/components/blocks/BlockProduct/BlockProductGrid'

import createFactory from '@/api/helpers/factory'
import productAttrs from '@/api/factories/product'

import '@/tests/unit/mocks/match-media'

describe('BlockProductGrid', () => {
  let previews

  beforeEach(() => {
    const products = createFactory(productAttrs, 12)
    const regularProduct = products.find((product) => product.in_stock)

    previews = regularProduct.previews
  })

  test('renders grid container', () => {
    const wrapper = shallowMount(BlockProductGrid, {
      propsData: {
        previews
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders grid container without video', () => {
    const products = createFactory(productAttrs, 12)
    const regularProduct = products.find((product) => product.in_stock)

    const wrapper = shallowMount(BlockProductGrid, {
      propsData: {
        previews: regularProduct.previews.filter(
          (item) => item.preview_type !== 'video'
        )
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
