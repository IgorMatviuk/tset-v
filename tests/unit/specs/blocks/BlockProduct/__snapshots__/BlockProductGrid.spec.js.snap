// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockProductGrid renders grid container 1`] = `
<div
  class="product-grid"
>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
</div>
`;

exports[`BlockProductGrid renders grid container without video 1`] = `
<div
  class="product-grid"
>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
  <div
    class="product-grid__item"
  >
    <blockproductpreview-stub
      item="[object Object]"
    />
  </div>
</div>
`;
